var/spack/repos/builtin/packages/kokkos-kernels/package.py

```diff
@@ -6,48 +6,261 @@
+
+            # Build Debug
+            if '+debug' in spec:
+                g_args.append('--debug')
+
+            # Backends
+            if '+serial' in spec:
+                g_args.append('--with-serial')
+            if '+openmp' in spec:
+                g_args.append('--with-openmp')
+            if '+pthreads' in spec:
+                g_args.append('--with-pthread')
+            if '+qthreads' in spec:
+                g_args.append('--with-qthreads=%s' % spec['qthreads'].prefix)
+            if '+cuda' in spec:
+                g_args.append('--with-cuda=%s' % spec['cuda'].prefix)
+            # Host architectures
+            host_arch = spec.variants['host_arch'].value
+            # GPU architectures
+            gpu_arch  = spec.variants['gpu_arch'].value
+            if host_arch != 'none':
+                arch_args.append(host_arch)
+            if gpu_arch != 'none':
+                arch_args.append(gpu_arch)
+            # Combined architecture flags
+            if arch_args:
+                g_args.append('--arch={0}'.format(','.join(arch_args)))
+
+            # CUDA options
+            if '+force_uvm' in spec:
+                cuda_options_args.append('force_uvm')
+            if '+use_ldg' in spec:
+                cuda_options_args.append('use_ldg')
+            if '+rdc' in spec:
+                cuda_options_args.append('rdc')
+            if '+enable_lambda' in spec:
+                cuda_options_args.append('enable_lambda')
+            if cuda_options_args:
+                g_args.append('--with-cuda-options={0}'
+                              .format(','.join(cuda_options_args)))
+
+            # Kokkos-kernels options
+            #if '+enable_eti' in spec:
+            #    kokkos_kernels_options_args.append('enable_eti')
+            #g_args.append('--with-options={0}'
+            #              .format(','.join(kokkos_kernels_options_args)))
+
+            # Kokkos options
+            if '+aggressive_vectorization' in spec:
+                kokkos_options_args.append('aggressive_vectorization')
+            if '+disable_profiling' in spec:
+                kokkos_options_args.append('disable_profiling')
+            if '+disable_dualview_modify_check' in spec:
+                kokkos_options_args.append('disable_dualview_modify_check')
+            if '+enable_profile_load_print' in spec:
+                kokkos_options_args.append('enable_profile_load_print')
+            if '+compiler_warnings' in spec:
+                kokkos_options_args.append('compiler_warnings')
+            if '+disable_deprecated_code' in spec:
+                kokkos_options_args.append('disable_deprecated_code')
+            if '+enable_eti' in spec:
+                kokkos_options_args.append('enable_eti')
+            if kokkos_options_args:
+                g_args.append('--with-kokkos-options={0}'
+                              .format(','.join(kokkos_options_args)))
```

Comment:
Do any of these with/enable/disable options have an inverse? Normally I like to add an `else` clause to explicitly disable things if they aren't needed.